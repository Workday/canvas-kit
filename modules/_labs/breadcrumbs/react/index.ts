import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
