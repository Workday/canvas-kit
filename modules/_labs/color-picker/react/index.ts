export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
