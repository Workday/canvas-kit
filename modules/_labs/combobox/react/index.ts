import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
