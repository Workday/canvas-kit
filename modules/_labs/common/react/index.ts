export * from './lib/Flex';
export * from './lib/Box';
