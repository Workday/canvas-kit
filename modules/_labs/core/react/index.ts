import type from './lib/type';
import space from './lib/space';
export default type;
export {type, space};
export * from './lib/type';
export * from './lib/StaticStates';
