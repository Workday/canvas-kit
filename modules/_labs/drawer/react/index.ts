import Drawer from './lib/Drawer';
import DrawerHeader from './lib/DrawerHeader';
export {Drawer, DrawerHeader};
export default Drawer;
export * from './lib/Drawer';
