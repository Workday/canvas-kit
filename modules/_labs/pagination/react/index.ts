import Pagination from './lib/Pagination';

export default Pagination;
export {Pagination};
export * from './lib/Pagination';
