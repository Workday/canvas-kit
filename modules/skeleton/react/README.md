# Canvas Kit Skeleton

A component that renders a skeleton (a placeholder for future content). The skeleton component takes
in children skeleton components and displays them with an animated sheen to indicate loading.

## Installation

```sh
yarn add @workday/canvas-kit-react
```

or

```sh
yarn add @workday/canvas-kit-react-skeleton
```

# Skeleton

This component places its children in a `div` marked with the `aria-hidden` attribute and displays
an `aria-live` status to specify that the content is loading.

## Usage

```tsx
import * as React from 'react';
import {
  Skeleton,
  SkeletonHeader,
  SkeletonShape,
  SkeletonText,
} from '@workday/canvas-kit-react-skeleton';

<Skeleton>
  <SkeletonShape width={50} height={50} borderRadius={99} />
  <SkeletonHeader />
  <h1>Hello!</h1>
  <SkeletonText lineCount={3} />
</Skeleton>;
```

> Note: In this example, non-Skeleton elements like `<h1>Hello!</h1>` will still show. This allows
> the flexibility for users to create more custom Skeletons.

## Static Properties

> None

## Component Props

> None

# SkeletonHeader

A component that renders a header placeholder for a skeleton. This component has a width of `100%`
and fixed height of `28px`.

## Usage

```tsx
import * as React from 'react';
import {Skeleton, SkeletonHeader} from '@workday/canvas-kit-react-skeleton';

<Skeleton>
  <SkeletonHeader />
</Skeleton>;
```

## Static Properties

> None

## Component Props

### Required

> None

### Optional

> None

# SkeletonShape

A component that renders a general shape placeholder for a skeleton. It gives the ability to alter
the borderRadius, width, and height to make various rectangular and circular shapes.

## Usage

```tsx
import * as React from 'react';
import {Skeleton, SkeletonShape} from '@workday/canvas-kit-react-skeleton';

<Skeleton>
  <SkeletonShape width={50} height={50} borderRadius={99} />
</Skeleton>;
```

## Static Properties

> None

## Component Props

### Required

> None

### Optional

#### `width: number | string`

> The width of the shape in `px` or `%`.

Default: `"100%"`

#### `height: number | string`

> The height of the shape in `px` or `%`.

Default: `"100%"`

#### `borderRadius: number | string`

> The borderRadius of the shape in `px` or `%`.

Default: `0`

# SkeletonText

A component that renders a text placeholder for a skeleton. Each line has a width of `100%` and a
fixed height of `21px`. If there is more then one line of text the last line of the SkeletonText
will have a width of `60%`.

## Usage

```tsx
import * as React from 'react';
import {Skeleton, SkeletonText} from '@workday/canvas-kit-react-skeleton';

<Skeleton>
  <SkeletonText lineCount={3} />
</Skeleton>;
```

## Static Properties

> None

## Component Props

### Required

> None

### Optional

#### `lineCount: number`

> The number of "lines" that it will display. If there's more than one line, then the last line will
> have a width of `60%`

Default: `2`
