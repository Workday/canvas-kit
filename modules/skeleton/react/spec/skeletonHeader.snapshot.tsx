import * as React from 'react';
import {create} from 'react-test-renderer';
import SkeletonHeader from '../lib/parts/skeletonHeader';

describe('SkeletonHeader', () => {
  test('default (no arguments)', () => {
    const subject = create(<SkeletonHeader />);

    expect(subject).toMatchSnapshot();
  });
});
