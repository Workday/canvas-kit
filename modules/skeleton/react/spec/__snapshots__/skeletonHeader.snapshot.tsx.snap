// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonHeader default (no arguments) 1`] = `
.emotion-0 {
  width: 100%;
  height: 28px;
  border-radius: 2px;
  background-color: #f0f1f2;
  margin-bottom: 16px;
}

<div
  className="emotion-0 emotion-1"
  height="28px"
  width="100%"
/>
`;
